irpc-iroh/src/lib.rs

```diff
@@ -215,52 +244,99 @@ impl<R: DeserializeOwned + Send + 'static> Iroh0RttProtocol<R> {
 }
 
 impl<R: DeserializeOwned + Send + 'static> ProtocolHandler for Iroh0RttProtocol<R> {
-    async fn on_connecting(&self, connecting: Connecting) -> Result<Connection, AcceptError> {
-        let (conn, _zero_rtt_accepted) = connecting
-            .into_0rtt()
-            .expect("accept into 0.5 RTT always succeeds");
-        Ok(conn)
-    }
-
-    fn accept(
-        &self,
-        connection: Connection,
-    ) -> impl std::future::Future<Output = Result<(), AcceptError>> + Send {
+    async fn on_accepting(&self, accepting: Accepting) -> Result<Connection, AcceptError> {
+        let zrtt_conn = accepting.into_0rtt();
         let handler = self.handler.clone();
         let request_id = self
             .request_id
             .fetch_add(1, std::sync::atomic::Ordering::AcqRel);
-        let fut = handle_connection(connection, handler).map_err(AcceptError::from_err);
-        let span = trace_span!("rpc", id = request_id);
-        Box::pin(fut.instrument(span))
+        handle_connection(&zrtt_conn, handler)
+            .map_err(AcceptError::from_err)
+            .instrument(trace_span!("rpc", id = request_id))
+            .await?;
```

Comment:
handle_connection loops until there is an error, and in that case returns an error, so we will exit here and everything below is unreachable, or am I wrong?